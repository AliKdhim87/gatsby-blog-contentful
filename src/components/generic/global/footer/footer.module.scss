.footer {
  display: flex;
  justify-content: center;
  margin-top: 3rem;
}
