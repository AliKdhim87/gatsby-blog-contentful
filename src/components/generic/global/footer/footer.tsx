import React from "react"
import { graphql, useStaticQuery } from "gatsby"

import { footer } from "./footer.module.scss"

const Footer: React.FC = () => {
  const {
    site: {
      siteMetadata: { author },
    },
  } = useStaticQuery(graphql`
    query {
      site {
        siteMetadata {
          title
          author
        }
      }
    }
  `)

  return (
    <footer className={footer}>
      <p>
        Created by {author}, Â©{new Date().getFullYear()}
      </p>
    </footer>
  )
}

export default Footer
