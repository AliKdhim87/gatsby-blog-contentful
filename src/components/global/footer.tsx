import React from "react"
import styled from "styled-components"
import { graphql, useStaticQuery } from "gatsby"

import { Header } from "semantic-ui-react"

const StyledFooter = styled.footer`
  width: 100%;
  min-height: 50px;
  background: ${({ theme }) => theme.colors.blue};
  display: flex;
  align-items: center;
  justify-content: center;
`

const Footer: React.FC = () => {
  const {
    site: {
      siteMetadata: { author },
    },
  } = useStaticQuery(graphql`
    query {
      site {
        siteMetadata {
          author
        }
      }
    }
  `)

  return (
    <StyledFooter>
      <Header as="p" inverted textAlign="center" size="small">
        Created by {author}, Â©{new Date().getFullYear()}
      </Header>
    </StyledFooter>
  )
}

export default Footer
