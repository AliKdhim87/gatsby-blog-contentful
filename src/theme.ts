import {DefaultTheme} from 'styled-components'

export enum ColorPalette {
  background = '#F3F4F5',
  blue = '#0E6EB8',
  red = '#FF546D',
  grey = '#f8f9fa',
  black = '#1B1C1D',
  white = '#FCFCFC',
  orange = '#FFAA50',
  darkBlack = '#000',
}

const baseTheme = {
  breakpoint: {
    mobile: '767px',
  },
  palette: {
    background: ColorPalette.background,
    blue: ColorPalette.blue,
    red: ColorPalette.red,
    grey: ColorPalette.grey,
    black: ColorPalette.black,
    white: ColorPalette.white,
    orange: ColorPalette.orange,
    darkBlack: ColorPalette.darkBlack,
  },
}

export const lightTheme: DefaultTheme = {
  ...baseTheme,
  textColor: ColorPalette.black,
  mainBg: ColorPalette.grey,
  secondary: ColorPalette.red,
  primary: ColorPalette.red,
  mainBgDark: ColorPalette.grey,
}
export const darkTheme: DefaultTheme = {
  ...baseTheme,
  textColor: ColorPalette.grey,
  mainBg: ColorPalette.black,
  secondary: ColorPalette.orange,
  primary: ColorPalette.orange,
  mainBgDark: ColorPalette.darkBlack,
}
