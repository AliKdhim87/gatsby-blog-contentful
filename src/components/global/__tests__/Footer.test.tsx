import React from 'react'
import renderer from 'react-test-renderer'

import {darkTheme, lightTheme} from 'theme'

import {PureFooter as Footer} from '../footer'

describe('Footer', () => {
  it('renders correctly and select the dark', () => {
    const tree = renderer
      .create(<Footer author="Ali Amouri Kadhim" color={darkTheme.textColor} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders correctly and select the light', () => {
    const tree = renderer
      .create(<Footer author="Ali Amouri Kadhim" color={lightTheme.textColor} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
