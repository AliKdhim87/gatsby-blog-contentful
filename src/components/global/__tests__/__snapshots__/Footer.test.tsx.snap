// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly and select the dark 1`] = `
<footer
  className="ui large inverted attached center aligned segment"
  style={
    Object {
      "marginTop": "1rem",
    }
  }
>
  <p
    className="ui tiny center aligned header"
    style={
      Object {
        "color": "#f8f9fa",
      }
    }
  >
    Created by 
    Ali Amouri Kadhim
     | ©
    2021
  </p>
</footer>
`;

exports[`Footer renders correctly and select the light 1`] = `
<footer
  className="ui large inverted attached center aligned segment"
  style={
    Object {
      "marginTop": "1rem",
    }
  }
>
  <p
    className="ui tiny center aligned header"
    style={
      Object {
        "color": "#1B1C1D",
      }
    }
  >
    Created by 
    Ali Amouri Kadhim
     | ©
    2021
  </p>
</footer>
`;
